common/persistence/cassandra/mutable_state_store.go

```diff
@@ -478,16 +478,38 @@ func (d *MutableStateStore) CreateWorkflowExecution(
 	return &p.InternalCreateWorkflowExecutionResponse{}, nil
 }
 
+// GetWorkflowExecution gets the workflow execution data from the cassandra DB.
+// If the request contains a runID it loads and returns the execution identified by that run ID.
+// If the request does not contain a runID, it first looks up the current run ID and then loads the corresponding execution object.
 func (d *MutableStateStore) GetWorkflowExecution(
 	ctx context.Context,
 	request *p.GetWorkflowExecutionRequest,
 ) (*p.InternalGetWorkflowExecutionResponse, error) {
+	runID := request.RunID
+	if runID == "" {
+		// get current runid using the permanentRunID record.
+		query := d.Session.Query(templateGetCurrentExecutionQuery,
+			request.ShardID,
+			rowTypeExecution,
+			request.NamespaceID,
+			request.WorkflowID,
+			permanentRunID,
+			defaultVisibilityTimestamp,
+			rowTypeExecutionTaskID,
+		).WithContext(ctx)
+
+		result := make(map[string]interface{})
+		if err := query.MapScan(result); err != nil {
+			return nil, gocql.ConvertError("GetWorkflowExecution", err)
+		}
+		runID = gocql.UUIDToString(result["current_run_id"])
```

Comment:
Can we reuse GetCurrentExecution() implementation here?

Reply:
Yes. We can. `GetCurrentExecution()` did a few additional deserialization of execution state. So I just took the runID part. I'll change this to just call `GetCurrentExecution()` instead.